src/main/java/frc/robot/scoreStates/ScoreStateContext.java

```diff
@@ -0,0 +1,28 @@
+package frc.robot.scoreStates;
+
+public class ScoreStateContext {
+    private static ScoreState currentState;
+    private static ScoreStateContext INSTANCE;
+
+    public ScoreStateContext getInstance() {
+        if (INSTANCE == null) {
+            INSTANCE = new ScoreStateContext();
+        }
+        return INSTANCE;
+    }
+
+    public void setCurrentState(ScoreState state) {
```

Comment:
Why not return commands for everything here? Just running these functions won't do anything anyway.
Also it'd be good to add the code that uses this class in robot container

Reply:
bad29dd

Reply:
fd842a3

Reply:
Also, could you add the code to robot container that actually uses this class? You added only the instantiation

Reply:
4a36e80
Not relevant to this PR, the button binding will be added later

Reply:
I'm not talking about button bindings, but about the issues raised in #39, e.g. https://github.com/Galaxia5987/robot-2024/pull/39#discussion_r1486123252. IMO, the state management is not versatile enough and could be improved. I have some idea on how to do that, but I'd like to see your implementation before I explain mine.

Reply:
The issue in #39 was resolved in Galaxia5987/cf3fee0

Reply:
Partly. If warming-up is not needed when the state changes, I don't see when will it happen if it does become necessary. because `prepareSubsystems()` is run only once.
And, I'm still not sure what you think of the issue I raised about Amping (you didn't say if I was wrong or agreed with it).

Reply:
prepareSubsystems happens continuously

Reply:
> prepareSubsystems happens continuously

But won't be added in this PR? What I ask for is the code that runs all of the scoring workflow. I think this PR is a good fit for it.
One last problem that I see is that calling `prepareSubsystems()` continuously might spam the gripper and elevator, in case the Amping direction is switched many times. Perhaps you don't care about that. (I know I suggested calling it continuously, but that's how it is, a better solution might solve one problem and raise another)